enhancements/network/routed-ingress-support-for-primary-udn-attached-vms-with-static-ips.md

```diff
@@ -0,0 +1,789 @@
++   Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+```
+
+The `IPAMClaim` status will have (at least) the following conditions:
+- IPAllocated: when the status is "True", it reports the IP address was
+  successfully allocated for the workload. The reason will be
+  `SuccessfulAllocation`. When the status is "False", it reports an IP
+  allocation error, and more details will be provided in the `reason`, and
+  `message` attributes.
+
+These conditions will look like - **successful** allocation:
+```
+type: IPAllocated
+status: "True"
+reason: SuccessfulAllocation
+message: "IP 192.168.1.5 allocated successfully"
+lastTransitionTime: "2025-05-13T11:56:00Z"
+```
+
+These conditions will look like - **failed** allocation:
+```
+type: IPAllocated
+status: "False"
+reason: IPAlreadyExists
+message: "Requested IP 192.168.1.5 is already assigned in the network"
+lastTransitionTime: "2025-05-13T11:56:00Z"
+```
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+The management cluster should have no reason to import an existing VM from
+another virtualization platform into OpenShift Virtualization.
+
+It makes more sense to import VMs into the hosted clusters, in order to provide
+tenants with routed ingress into their hosted cluster workloads.
+
+There should be no hypershift platform-specific considerations with this
+feature.
+
+#### Standalone Clusters
+
+Full support.
+
+#### Single-node Deployments or MicroShift
+
+Full support; there might be some increased resource usage when scaling up
+number of VMs requesting IP and MAC address, since it will mean more processing
+in OVN-Kubernetes, and more CRs to reconcile.
+
+### Implementation Details/Notes/Constraints
+
+TODO
+
+### Risks and Mitigations
+
+TODO
+
+### Drawbacks
+
+The biggest drawback we have to implementing this feature is lack of clear
+asks from customers - we're pretty much guessing what they would want to use.
+
+We do not know (for instance):
+- is this feature only for VMs ?
+- is this feature only about importing VMs ? Should we allow creating new VMs
+  with dedicated MAC / IP / gateway requests ?
+- are the end users (admins) after a centralized IP management alternative ?
+  Do they want to have a single place to check their network address
+  assignments ?
+
+## Alternatives (Not Implemented)
+
+### Requesting IP addresses on a VM's interfaces
+An option which would provide a simpler implementation (and better / more
+explicit UX) would be to modify the VMI specification, adding an attribute for
+the user (or the tool templating the VM - e.g. MTV) to specify which IP
+addresses they want the VM's interface to take.
+
+It would be updating the
+[v1.Interface](https://kubevirt.io/api-reference/main/definitions.html#_v1_interface)
+with an ipAddresses array. This attribute is defined in
+`VMI.Spec.Domain.Devices.Interfaces`.
+
+This option (despite being the simplest to implement) is not taken into
```

Comment:
Is this is a feasible future goal? If in the future we would like to provide a proper API for new VMs to specify the IP it seems like a good idea to do it here.